src/Equinox.CosmosStore/CosmosStore.fs

```diff
@@ -615,7 +622,18 @@ module internal Tip =
         | ReadResult.NotFound -> return Result.NotFound
         | ReadResult.Found tip ->
             let minIndex = maybePos |> Option.map (fun x -> x.index)
-            return Result.Found (Position.fromTip tip, tip.i, Enum.EventsAndUnfolds(tip, ?maxIndex = maxIndex, ?minIndex = minIndex) |> Array.ofSeq) }
+            return Result.Found (Position.fromEtagAndIndex (tip._etag, tip.n), tip.i, Enum.EventsAndUnfolds(tip, ?maxIndex = maxIndex, ?minIndex = minIndex) |> Array.ofSeq) }
+    let tryFindOrigin (tryDecode: ITimelineEvent<EventBody> -> 'event voption, isOrigin: 'event -> bool) xs =
+        let stack = ResizeArray()
+        let isOrigin' (u: ITimelineEvent<EventBody>) =
+            match tryDecode u with
+            | ValueNone -> false
+            | ValueSome e ->
+                stack.Insert(0, e) // WalkResult always renders events ordered correctly - here we're aiming to align with Enum.EventsAndUnfolds
+                isOrigin e
+        match xs |> Seq.tryFindBack isOrigin' with
+        | Some x -> ValueSome (x, stack.ToArray())
+        | None -> ValueNone
```

Comment:
```suggestion
        | Some x -> (true, stack.ToArray())
        | None -> (false, [||])
```

Reply:
As per other comment, we use it in the next PR